wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift

```diff
@@ -112,20 +127,118 @@ extension ConversationListViewController {
+            filter: .groups,
+            isSelected: selectedFilter == .groups
+        )
+        let oneToOneConversationsAction = createFilterAction(
+            title: FilterMenuLocale.OneOnOneConversations.title,
+            filter: .oneToOneConversations,
+            isSelected: selectedFilter == .oneToOneConversations
+        )
+
+        // Create the menu
+        let filterMenu = UIMenu(
+            children: [
+                allConversationsAction,
+                favoritesAction,
+                groupsAction,
+                oneToOneConversationsAction
+            ]
+        )
+
+        // Create the filter button and assign the menu
+        let filterButton = UIButton(type: .system)
+        filterButton.setImage(selectedFilterImage, for: .normal)
+        filterButton.showsMenuAsPrimaryAction = true
+        filterButton.menu = filterMenu
+
+        navigationItem.rightBarButtonItems?.append(UIBarButtonItem(customView: filterButton))
+    }
+
+    private func createFilterAction(
+        title: String,
+        filter: FilterType,
+        isSelected: Bool
+    ) -> UIAction {
+        let imageName = getFilterImageName(for: filter, isSelected: isSelected)
+
+        let font = UIFont.systemFont(ofSize: 17)
+        let configuration = UIImage.SymbolConfiguration(font: font)
+        let actionImage = UIImage(systemName: imageName)?.applyingSymbolConfiguration(configuration)
+
+        // Apply the tint color conditionally based on the selection state
+        let tintedActionImage: UIImage?
+        if isSelected {
+            tintedActionImage = actionImage?.withTintColor(UIColor.accent(), renderingMode: .alwaysOriginal)
+        } else {
+            tintedActionImage = actionImage
+        }
+
+        let titleAttributes: [NSAttributedString.Key: Any] = [
+            .foregroundColor: isSelected ? UIColor.accent() : SemanticColors.Label.textDefault
+        ]
+
+        let attributedTitle = NSAttributedString(string: title, attributes: titleAttributes)
+
+        let action = UIAction(title: title, image: tintedActionImage) { [weak self] _ in
+            self?.selectedFilter = filter
+            self?.setupRightNavigationBarButtons()
+        }
+
+        action.setValue(attributedTitle, forKey: "attributedTitle")
+
+        return action
+    }
+
+    private func getFilterImageName(for filter: FilterType, isSelected: Bool) -> String {
+        switch filter {
+        case .allConversations:
+            return isSelected ? FilterImageName.textBubbleFill.rawValue : FilterImageName.textBubble.rawValue
+        case .favorites:
+            return isSelected ? FilterImageName.starFill.rawValue : FilterImageName.star.rawValue
+        case .groups:
+            return isSelected ? FilterImageName.person3Fill.rawValue : FilterImageName.person3.rawValue
+        case .oneToOneConversations:
+            return isSelected ? FilterImageName.personFill.rawValue : FilterImageName.person.rawValue
```

Comment:
Maybe one can return the right type and do the `.rawValue` once on the caller?

Reply:
Done! https://github.com/wireapp/wire-ios/commit/10529635d8ea8a54a6de63e16b863a87718d30b7